src/value.rs

```diff
@@ -198,15 +217,34 @@ impl ValueSum {
             sign,
         )
     }
+
+    pub(crate) fn into<V: TryFrom<i64>>(self) -> Result<V, BuildError> {
```

Comment:
_From str4d ([r2542756837](https://github.com/zcash/orchard/pull/471#discussion_r2542756837))_

Rename this; we should never have a manual `into()` method that aliases the `Into` trait while using a different method signature (which is closer to `TryInto::try_into`).

```suggestion
    pub(crate) fn into_value_balance<V: TryFrom<i64>>(self) -> Result<V, BuildError> {
```

Reply:
_From str4d ([r2543702226](https://github.com/zcash/orchard/pull/471#discussion_r2543702226))_

In a pairing with daira, we also noted that currently [ZIP 230](https://zips.z.cash/zip-0230#orchardzsa-asset-burn-description) defines `valueBurn: u64`, but allowing that full range now is annoying because every other value balance is an `i64`, and it would force us to add a bunch of extra handling logic that would only be used for burns (and would be incompatible with future asset-specific value balances when we eventually have a subsequent pool that supports ZSAs via a turnstile).

For simplicity, we should update ZIP 230 to only allow burn amounts that fit into both a `u64` and an `i64`, i.e. `u63`.

Reply:
FIXED, TODO